.github/actions/release/action.yml

```diff
@@ -10,51 +13,177 @@ inputs:
+        echo "::notice::Configuring git."
         git config --global user.email "release@centreon.com"
         git config --global user.name "Centreon"
 
-        if [ -z "$OLDV" ]; then
-          echo "No existing version, starting at $MAJOR_VERSION.0"
-          NEW_VERSION="$MAJOR_VERSION.0"
-          git tag -a "centreon-${{ inputs.module_name }}-$NEW_VERSION" -m "version $NEW_VERSION"
-          git push --follow-tags
-        else
-          OLD_MINOR_VERSION=$(echo $OLDV | grep -oP '([0-9]+$)')
-          NEW_MINOR_VERSION=$(echo $((OLD_MINOR_VERSION + 1)))
-          NEW_VERSION=$MAJOR_VERSION.$NEW_MINOR_VERSION
-          git tag -a "centreon-${{ inputs.module_name }}-$NEW_VERSION" -m "version $NEW_VERSION"
-          git push --follow-tags
-        fi
+        # Create release tags on git for each release components
+        echo "::notice::Creating release tags."
+        for TAG in ${NEW_STABLE_TAGS[@]}; do
+          if [ -z $(git tag --list "$TAG" | head -n 1) ]; then
+            git tag -a "$TAG" -m "$TAG"
+            git push --follow-tags
+            echo "::notice::Tagging branch with $TAG."
+          else
+            echo "::error::Release tag $TAG already exists, exiting."
+            exit 1
+          fi
+        done
+      shell: bash
 
-        if [ "$HOTFIX" == "hotfix" ]; then
-          TYPE=Hotfix
+    - name: Create GITHUB releases from new release tags
+      run: |
+        # Install gh cli
+        echo "::notice::Installing GH CLI."
+        if ! command -v gh &> /dev/null; then
+          echo "Installing GH CLI."
+          type -p curl >/dev/null || (sudo apt update && sudo apt install curl -y)
+          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+          sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+          sudo apt update
+          sudo apt install gh -y
         else
-          TYPE=Release
+          echo "GH CLI is already installed."
         fi
 
-        VERSION_DATA="{\"archived\":false,\"releaseDate\":\"$(date +%Y-%m-%d)\",\"name\":\"centreon-${{ inputs.module_name }}-$NEW_VERSION\",\"description\":\"$TYPE:$RELEASE_ID\",\"projectId\":${{ inputs.jira_project_id }},\"released\":false}"
+        # Create GITHUB release for each release components
+        echo "::notice::Creating GITHUB releases."
+        for TAG in ${NEW_STABLE_TAGS[@]}; do
+          echo "::notice::Creating release with title $TAG for tag $TAG."
+          gh release create $TAG --target "${{ inputs.github_ref_name }} --title "$TAG" --verify-tag
+        done
+      shell: bash
+
+    - name: Create stable JIRA versions from new release tags
+      run: |
+        set -eux
```

Comment:
do you need all those flags ?

Reply:
actually they were quite useful to troubleshoot potentially empty variables (especially tags)
the could be removed (at least the -x , -eu are nice to have anytime)